### DISCUSSION TITLE
Create domain in sequel

### DISCUSSION THREAD

**Discussion by marek22k** [jeremyevans/sequel#2054] [General]
Hello,

I would like to know how to enter the following SQL statement in Sequel:
```
CREATE DOMAIN asn AS bigint
   CHECK(VALUE >= 0 AND VALUE <= 4294967295);
```
(as I couldn't find a unsigned 32bit integer)

I could not find a create_domain function in the documentation.

I would be very glad to get some answers!

**Comment by jeremyevans** [comment]
There isn't a method for creating domains in Sequel.  You should use:

```ruby
DB.run("CREATE DOMAIN asn AS bigint CHECK(VALUE >= 0 AND VALUE <= 4294967295);")
```

**Reply by marek22k** [reply]
Thank you!

**Reply by marek22k** [reply]
I would have one more question: The type, when packed into an array, is not automatically converted to an integer, but returned as a Ruby string.
For the array, for example, I had to do the following:
```ruby
DB.register_array_type :asn
```
Do I have to do something like this to register the type as bigint (like `DB.register_integer :asn`)?

**Reply by jeremyevans** [reply]
See the register_array_type documentation: https://sequel.jeremyevans.net/rdoc-plugins/classes/Sequel/Postgres/PGArray/DatabaseMethods.html#method-i-register_array_type .  You probably want to pass :scalar_typecast and :converter.

**Reply by marek22k** [reply]
Thanks for the answer. Why do I have to specify the two? Shouldn't it be enough if I specify the type?
```ruby
@db.register_array_type :asn, converter: -> (element) { element.to_i }
```
converts it to integers, but
```ruby
@db.register_array_type :asn, type_symbol: :integer
```
doesn't convert it

**Reply by jeremyevans** [reply]
`:type_symbol` is used for typecasting (`Database#typecast_value`), not for converting while parsing values received from the database.
